commands/credentials.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+const co = require('co')
+const cli = require('heroku-cli-util')
+
+function * run (context, heroku) {
+  const fetcher = require('../lib/fetcher')(heroku)
+
+  const {app, args, flags} = context
+
+  let showCredentials = co.wrap(function * () {
+    let db = yield fetcher.database(app, args.database)
+    cli.log(`Connection info string:
+   "dbname=${db.database} host=${db.host} port=${db.port || 5432} user=${db.user} password=${db.password} sslmode=require"
+Connection URL:
+   ${db.url.href}`)
+  })
+
+  let reset = co.wrap(function * () {
+    const host = require('../lib/host')
+    let db = yield fetcher.addon(app, args.database)
+    yield cli.action(`Resetting credentials on ${cli.color.addon(db.name)}`, co(function * () {
+      yield heroku.post(`/client/v11/databases/${db.name}/credentials_rotation`, {host: host(db)})
+      // TODO: is the promotion stuff necessary here?
```

Comment:
I don't think so, but what promotion stuff? If you mean ensuring the app gets the new config vars, no, Shogun (or Yobuko) will post them to the app and the app will restart. Because `DATABASE_URL` is now a real attachment, that will pick up the new value as well.

If dealing with a legacy app where `DATABASE_URL` was just copy-and-pasted, I *think* API has (had?) some magic to automatically update config vars matching the same value, but I'm not sure about this. There also seems to be an implicit attachment behavior with `config:set` matching a config var right now, so I can't think of a way to test this directly.